File: example/java-lambda/src/main/java/com/lamblin/example/lambda/controller/TodoController.java
package com.lamblin.example.lambda.controller;

import static com.lamblin.core.model.HttpResponse.ok;

import java.util.List;

import com.google.common.collect.ImmutableList;
import com.lamblin.core.model.HttpMethod;
import com.lamblin.core.model.HttpResponse;
import com.lamblin.core.model.annotation.Controller;
import com.lamblin.core.model.annotation.Endpoint;
import com.lamblin.core.model.annotation.PathParam;
import com.lamblin.core.model.annotation.QueryParam;
import com.lamblin.core.model.annotation.RequestBody;
import com.lamblin.example.lambda.model.TodoList;
import com.lamblin.example.lambda.service.DataService;

@Controller
public class TodoController {

  private static final String TODOS_ROOT_ENDPOINT = "/todo-lists";
  private static final String TODOS_SPECIFIC_LIST = "/todo-lists/{uid}";

  private final DataService dataService;

  public TodoController(DataService dataService) {
    this.dataService = dataService;
  }

  @Endpoint(path = TODOS_ROOT_ENDPOINT, method = HttpMethod.GET)
  public HttpResponse<List<TodoList>> getAllTodoLists() {
    return ok(dataService.fetchAll());
  }

  @Endpoint(path = TODOS_ROOT_ENDPOINT, method = HttpMethod.POST)
  public HttpResponse<List<TodoList>> createTodoList(@RequestBody TodoList todoList) {
    return ok(ImmutableList.<TodoList>builder()
        .addAll(dataService.fetchAll())
        .add(todoList)
        .build());
  }

  @Endpoint(path = TODOS_ROOT_ENDPOINT, method = HttpMethod.GET)
  public HttpResponse<TodoList> getTodoListByName(@QueryParam("name") String name) {
    return ok(dataService.fetchListForName(name));
  }

  @Endpoint(path = TODOS_SPECIFIC_LIST, method = HttpMethod.GET)
  public HttpResponse<TodoList> getSpecificTodoList(@PathParam("uid") String uid) {
    return ok(dataService.fetchList(uid));
  }

  @Endpoint(path = TODOS_SPECIFIC_LIST, method = HttpMethod.DELETE)
  public HttpResponse<Void> deleteTodoList(@PathParam("uid") String uid) {
    dataService.deleteList(uid);

    return ok();
  }

}

File: example/kotlin-lambda/src/main/kotlin/com/lamblin/example/lambda/controller/TodoController.kt
package com.lamblin.example.lambda.controller

import com.lamblin.core.model.HttpMethod
import com.lamblin.core.model.HttpResponse
import com.lamblin.core.model.annotation.*
import com.lamblin.example.lambda.model.TodoList
import com.lamblin.example.lambda.service.DataService

const val TODOS_ROOT_ENDPOINT = "/todo-lists"
const val TODOS_SPECIFIC_LIST = "/todo-lists/{uid}"

@Controller
class TodoController(private val dataService: DataService) {

    @Endpoint(TODOS_ROOT_ENDPOINT, HttpMethod.GET)
    fun getAllTodoLists() = HttpResponse.ok(dataService.fetchAll())

    @Endpoint(TODOS_ROOT_ENDPOINT, HttpMethod.POST)
    fun createTodoList(@RequestBody todoList: TodoList) = HttpResponse.ok(dataService.fetchAll() + todoList)

    @Endpoint(TODOS_ROOT_ENDPOINT, HttpMethod.GET)
    fun getTodoListByName(@QueryParam("name") name: String) = HttpResponse.ok(dataService.fetchListForName(name))

    @Endpoint(TODOS_SPECIFIC_LIST, HttpMethod.GET)
    fun getSpecificTodoList(@PathParam("uid") uid: String) = HttpResponse.ok(dataService.fetchList(uid))

    @Endpoint(TODOS_SPECIFIC_LIST, HttpMethod.DELETE)
    fun deleteTodoList(@PathParam("uid") uid: String): HttpResponse<Void> {
        dataService.deleteList(uid)

        return HttpResponse.ok()
    }

}